pleasew

```diff
@@ -42,19 +42,27 @@ fi
 
 # Find the os / arch to download. You can do this quite nicely with go env
 # but we use this script on machines that don't necessarily have Go itself.
-OS=`uname`
-if [ "$OS" = "Linux" ]; then
-    GOOS="linux"
-elif [ "$OS" = "Darwin" ]; then
-    GOOS="darwin"
-else
-    echo -e >&2 "${RED}Unknown operating system $OS${RESET}"
-    exit 1
-fi
-# Don't have any builds other than amd64 at the moment.
-ARCH="amd64"
+case "`uname`" in
+  Linux)
+    GOOS="linux";;
+  Darwin)
+    GOOS="darwin";;
+  *)
+    echo "Unknown operating system $(uname)"
+    exit 1;;
+esac
+
+case "`uname -m`" in
+  arm*)
```

Comment:
on a raspberry pi with a 64-bit kernel enabled, we get 
```
$ uname -m
aarch64
$ uname -a
Linux xxxxx 5.10.17-v8+ #1421 SMP PREEMPT Thu May 27 14:01:37 BST 2021 aarch64 aarch64 aarch64 GNU/Linux
```
aarch64 supposedly is the same as arm64 (https://stackoverflow.com/a/47274698) so I think this could be expanded
to

```bash
case "`uname -m`" in
  arm* | aarch*)
     GOARCH="arm64" ;;
...
``` 
if that makes sense to you too :)

Reply:
Yeah +1, it's one of those things like amd64 / x86_64. Naming is hard...